hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/server/api/protocolrecords/FederationQueueWeight.java

```diff
@@ -170,6 +170,26 @@ public static void checkHeadRoomAlphaValid(String headRoomAlpha) throws YarnExce
     }
   }
 
+  public static void checkPolicyManagerValid(String policyManager) throws YarnException {
+    switch (policyManager) {
+      case "org.apache.hadoop.yarn.server.federation.policies.manager.HashBroadcastPolicyManager":
+        throw new YarnException("HashBroadcastPolicyManager does not support the use of queue weights.");
+      case "org.apache.hadoop.yarn.server.federation.policies.manager.HomePolicyManager":
+        throw new YarnException("HomePolicyManager does not support the use of queue weights.");
```

Comment:
I think we can do a list for the ones that support weights, and if it's not there, return the exception,

Reply:
I will improve this part of the code.